src/wasm-v8.cc

```diff
@@ -576,62 +531,62 @@ auto ValType::make(ValKind k) -> own<ValType> {
       // TODO(wasm+): support new value types
       assert(false);
   };
-  auto result = seal<ValType>(valtype);
-  stats.make(Stats::VALTYPE, result);
-  return own<ValType>(result);
+  stats.make(Stats::VALTYPE, valtype);
+  return own<ValType>(valtype);
 }
 
 auto ValType::copy() const -> own<ValType> {
   return make(kind());
 }
 
 auto ValType::kind() const -> ValKind {
-  return impl(this)->kind;
+  return ValTypeImpl::from(this)->kind;
 }
 
 
 // Extern Types
 
-struct ExternTypeImpl {
+struct ExternTypeKind {
```

Comment:
I'm confused, why do we need this aux wrapper?

Reply:
I wrote the answer in response to a different comment: https://github.com/WebAssembly/wasm-c-api/pull/161#discussion_r520960347

Reply:
So IIUC, you are saying that this (and the static_asserts) could be avoided if we used virtual inheritance on the Impl classes? As long as that does not leak to the .hh file and imposes no other overhead, I suppose I'd be fine with that.

Reply:
No, `FuncType` etc. would need to virtually inherit from `ExternType` too. Both inheritance edges need to be marked virtual for the bases to be merged.

https://isocpp.org/wiki/faq/multiple-inheritance#virtual-inheritance-where